test/e2e/qat/qatplugin_dpdk.go

```diff
@@ -165,6 +182,52 @@ func describeQatDpdkPlugin() {
 			ginkgo.It("does nothing", func() {})
 		})
 	})
+
+	ginkgo.Context("When QAT's autoreset is off [Functionality:heartbeat]", func() {
+		ginkgo.BeforeEach(func() {
+			ginkgo.By("creating a configMap before plugin gets deployed")
+			e2ekubectl.RunKubectlOrDie(f.Namespace.Name, "create", "configmap", "--from-literal", "qat.conf=AutoresetEnabled=off", "qat-config")
+		})
+
+		ginkgo.It("checks if the heartbeat is read correctly", func(ctx context.Context) {
+			injectError(ctx, f, nodeName)
+
+			ginkgo.By("seeing if there is zero resource")
+			if err := utils.WaitForNodesWithResource(ctx, f.ClientSet, resourceName, 100*time.Second, utils.WaitForZeroResource); err != nil {
+				framework.Failf("unable to wait for nodes to have no resource: %v", err)
+			}
+		})
+	})
+
+	ginkgo.Context("When QAT's autoreset is on [Functionality:auto-reset]", func() {
+		ginkgo.BeforeEach(func() {
+			ginkgo.By("creating a configMap before plugin gets deployed")
+			e2ekubectl.RunKubectlOrDie(f.Namespace.Name, "create", "configmap", "--from-literal", "qat.conf=AutoresetEnabled=on", "qat-config")
+		})
+
+		ginkgo.It("checks if an injected error gets solved", func(ctx context.Context) {
+			injectError(ctx, f, nodeName)
+
+			ginkgo.By("seeing if there is zero resource")
+			if err := utils.WaitForNodesWithResource(ctx, f.ClientSet, resourceName, 100*time.Second, utils.WaitForZeroResource); err != nil {
+				// It should not fail since there can be a case auto-reset function worked before plugin notices.
+				// It is still useful to wait for some time to ensure that injected error affects other specs.
```

Comment:
Umm unless I did not understand the meaning of "spec" in ginkgo... it just means the other parts of the flow?
(https://onsi.github.io/ginkgo/#writing-specs)

IMO, the matter you are suggesting is not related to this part. Because, this part is about `auto-reset on`. Testing if there is zero resource is the work that should be done in `auto-reset off` and `heartbeat`..

If you do not like this, then we need to inject the error before auto-reset is on, then it should be somewhere in `JustBeforeEach` which looks quite bad in my perspective.

Reply:
> Testing if there is zero resource is the work that should be done in `auto-reset off` and `heartbeat`..

That's what I'm thinking too. I was wondering why we then have to wait for "zero resource" here. Can you add more comments in the code explaining this.

Reply:
Ah, yeah, good point. let me add comments, but it is necessary in my opinion.